### PR TITLE
PyDbg thread now works without start command

### PR DESCRIPTION
**Pull request by jtpereyda**
Issue #237 

Needs confirmation of functionality.

### CODE REVIEW

File: boofuzz/utils/debugger_thread_pydbg.py [hunk 1]
```diff
@@ -115,47 +113,47 @@ def run(self):
...
-                elif "Access is denied." in str(e):
-                    self.process_monitor.log("It may be that your process died before it could be attached.")
-            self.finished_starting.set()
-            self.dbg.run()
-            self.process_monitor.log("debugger thread-%s exiting" % self.getName())
-
-            # TODO: removing the following line appears to cause some concurrency issues.
-            # del self.dbg
+        if len(self.start_commands) > 0 and self.proc_name is not None:
+            self.spwan_target()
```
Comment (KevinCooper) [inline boofuzz/utils/debugger_thread_pydbg.py]: spawn_target*

File: boofuzz/utils/debugger_thread_pydbg.py [hunk 2]
```diff
@@ -115,47 +113,47 @@ def run(self):
...
+            self.watch()
+        elif len(self.start_commands) > 0:
+            self.spwan_target()
```
Comment (KevinCooper) [inline boofuzz/utils/debugger_thread_pydbg.py]: spawn_target*

File: boofuzz/utils/debugger_thread_pydbg.py [hunk 3]
```diff
@@ -96,10 +96,8 @@ def _dbg_callback_user(self, dbg):
 
         return pydbg.defines.DBG_CONTINUE
 
-    def spawn_target(self):
-        # TODO move spawn_target into run to remove the half-initialized state between calling the two functions
-        # TODO checks: debugger thread already active; process already started; no start_commands
-
+    def spwan_target(self):
```
Comment (KevinCooper) [inline boofuzz/utils/debugger_thread_pydbg.py]: spawn_target

**Comment by KevinCooper** [comment]
This corrects the issue.  Potential spelling mistake with the `spwan_target`

**Comment by jtpereyda** [comment]
@KevinCooper Thank you for the test and review!

**Pull request merged**